src/pages/GeneralAdminView/NewRestaurantForm/NewRestaurantForm.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState }from 'react';
+import axios from 'axios';
+import Cookies from 'universal-cookie';
+import { useSelector } from 'react-redux';
+import { toast } from "react-toastify";
+import './NewRestaurantForm.css';
+import languageSelector from '../../../assets/languages/languageSelector';
+
+const NewRestaurantForm = () => {
+  const cookies = new Cookies();
+  const [restaurant_name, setRestautant_name] = useState('');
+  const [errors, setErrors] = useState({});
+  const [logo, setLogo] = useState(null);
+  const [mainPhoto, setMainPhoto] = useState(null);
+  const [adminEmail, setAdminEmail] = useState('');
+  const language = useSelector(state=> state.languageReducer);
+  const apiUrl = process.env.REACT_APP_API_URL;
+  const jwtToken = cookies.get('token');
+  const config = {
+    headers: {
+      Authorization: `Bearer ${jwtToken}`,
+    },
+  };
+
+  const validateFields = () => {
+    const validationErrors = {};
+
+    if (!restaurant_name.trim()) {
+      validationErrors.restaurant_name = languageSelector(language, 'newRestaurantFormNameError');
+    }
+    if (!logo) {
+      validationErrors.logo = languageSelector(language, 'newRestaurantLogoError');
+    }
+    if (!mainPhoto) {
+      validationErrors.mainPhoto = languageSelector(language, 'newRestaurantMainPhotoError');;
+    }
+    if (!adminEmail.trim() || !/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/.test(adminEmail)) {
```

Comment:
Move this regex to a constant to be imported through the project

Reply:
Resolved